src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs

```diff
@@ -878,5 +878,93 @@ private void MinerCreateNewBlockValidity()
             //    SetMockTime(0);
             //    mempool.clear();
         }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockCheckNodeConsensusTipIsCorrect()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode miner = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+
+                miner.WithWallet();
+                builder.StartAll();
+                
+                TestHelper.MineBlocks(miner, 5);
+
+                Assert.Equal(5, miner.FullNode.ConsensusManager().Tip.Height);
+            }
+        }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockCheckPeerHasNewBlock()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode miner = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+                
+                CoreNode[] syncers = new CoreNode[]
+                {
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD(),
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD(),
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD()
+                };
+
+                miner.WithWallet();
+                builder.StartAll();
+                
+                foreach (CoreNode syncer in syncers)
+                {
+                    miner.CreateRPCClient().AddNode(syncer.Endpoint, true);
+                }
+
+                Assert.True(syncers.All(x => x.FullNode.ConsensusManager().Tip.Height == 0));
+
+                TestHelper.MineBlocks(miner, 3);
+
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[0]));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[1]));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[2]));
+
+                Assert.True(syncers.All(x => x.FullNode.ConsensusManager().Tip.Height == 3));
+            }
+        }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockNotPushedToConsensusCode_SupercededByBetterBlockOnReorg_InitialBlockRejected()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode node1 = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+                CoreNode node2 = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+
+                node1.WithWallet();
+                node2.WithWallet();
+
+                builder.StartAll();
+                
+                node1.CreateRPCClient().AddNode(node2.Endpoint, true);
```

Comment:
Use `TestHelper.ConnectAndSync` here.

Reply:
Actually you shouldn't connect here yet.

Reply:
I have to connect and sync to get a common 5 block chain between the nodes.